rulesets_command.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+use SilverStripe\SupportedModules\MetaData;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Command\Command;
+
+$rulesetsCommand = function(InputInterface $input, OutputInterface $output): int {
+    // This is the code that is executed when running the 'rulesets' command
+
+    // Variables
+    global $OUT, $REPOS_WITH_RULESETS_UPDATED;
+    $OUT = $output;
+
+    // Validate system is ready
+    validate_system();
+
+    // Modules
+    $modules = [];
+    $modulesCurrentMajor = filtered_modules(MetaData::HIGHEST_STABLE_CMS_MAJOR, $input);
+    $modulesPreviousMajor = filtered_modules(MetaData::HIGHEST_STABLE_CMS_MAJOR - 1, $input);
+    foreach ([$modulesCurrentMajor, $modulesPreviousMajor] as $modulesList) {
+        foreach ($modulesList as $module) {
+            // Important! Only include modules on the "silverstripe" account
+            if ($module['account'] !== 'silverstripe') {
+                continue;
+            }
+            $modules[] = $module;
+        }
+    }
+
+    // Update rulesets
+    foreach ($modules as $module) {
+        $account = $module['account'];
+        $repo = $module['repo'];
+
+        // Fetch existing rulesets
+        $rulesets = github_api("https://api.github.com/repos/$account/$repo/rulesets");
+        $branchRulesetID = 0;
+        $tagRulesetID = 0;
+        foreach ($rulesets as $ruleset) {
+            $id = $ruleset['id'];
+            $name = $ruleset['name'];
+            if ($name === BRANCH_RULESET_NAME) {
+                $branchRulesetID = $id;
+            }
+            if ($name === TAG_RULESET_NAME) {
+                $tagRulesetID = $id;
+            }
+        }
+
+        // Get any additional branches to add
+        // Assumption is that if the default branch is main/master, then the repo uses
+        // a non-numeric style branching system (e.g. main, master) and that needs to be protected
+        // [0-9]* branch protection will still be applied, on the chance that the repo is converted
+        // to uses a numeric style branch system in the future and we would want branch protection
+        // to start immediately on the new branches
+        $additionalBranchConditions = [];
+        $defaultBranch = github_api("https://api.github.com/repos/$account/$repo")['default_branch'];
+        if (in_array($defaultBranch, ['main', 'master'])) {
+            $additionalBranchConditions[] = "refs/heads/$defaultBranch";
+        }
+
+        // Create rulesets
+        // Note: This will read from the "rulesets" directory
+        // In each of those json rulesets there is "bypass_actors"."actor_id" = 5
+        // This translates to the "Repository admin" role
```

Comment:
I didn't find one. Instead I simply manually tested the different bypass levels to see if the github-action user could bypass or not

Note you get these by default + some teams linked to the repo

![image](https://github.com/silverstripe/module-standardiser/assets/4809037/ac581b60-e38f-4b0d-9050-f75682025e5e)

"Deal with that when it happens" is fine IMO, using this PR we can very easily change the bypass to maintainer if desired

Reply:
> I didn't find one. Instead I simply manually tested the different bypass levels to see if the github-action user could bypass or not

Gotta love when APIs require magic values but don't document those values lol.

> "Deal with that when it happens" is fine IMO, using this PR we can very easily change the bypass to maintainer if desired

My concern isn't in how difficult it would be to change later. My concern is that if we're forced to change it later we're effectively escalating the permissions for roles we don't want to escalate the permissions for. Community peer reviewers are probably gonna be at the maintainer level, I'd think.

Reply:
Seems like we're then basically forced to make any new CMS squad members "Repository Admin" in that case so that they can merge PRs normally

Yeah that's probably not ideal

Having a look at https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization - what's there than would neccesitate 'maintain' over 'admin' for community peer reviewers?

Seems like it might be better to make the reviewers "Write" and then set the bypass on the rulesets to "Maintain" instead of "Repository admin"

Reply:
Oh nice, didn't know that matrix existed.
Yeah I forgot about "write" access lol. That will be perfect for reviewers. No worries then, leave as is. I'll make sure the card for creating the reviewer role explicitly gives them write access only.